### ISSUE TITLE
How to set keep_alive_timeout using JDBC 0.4.2?

### ISSUE THREAD

**Issue by shashank-gb** [ClickHouse/clickhouse-java#1348]
Hi,

Some days back, I was using `clickhouse-jdbc-0.2.6-shaded`. In that, there is a method `setKeepAliveTimeout(int keepAliveTimeout)` which is deprecated and I was using the same to set the `keep_alive_timeout` config value.

Now I am shifted to `clickhouse-jdbc-0.4.2-shaded`. Here `ClickhouseProperties` itself removed. Properties is being used Instead.
My question is, How to set the `keep_alive_timeout` in `clickhouse-jdbc-0.4.2-shaded` ? and I want to print all the clickhouse properties key-value pair, how to do that ? previously, I was using `asProperties()` method which is a member with the `ClickhouseProperties` class.

In short,
1. How to set `keep_alive_timeout` using `clickhouse-jdbc-0.4.2-shaded` ?
2. What is the alternative of `asProperties()` in `clickhouse-jdbc-0.4.2-shaded` ?
3. How to find the replacement of deprecated members ?

Thanks

**Comment by zhicwu** [comment]
Hi @shashank-gb, there's a [cheat sheet](/ClickHouse/clickhouse-java/tree/main/clickhouse-jdbc#upgrade-path) to start with for upgrading from legacy driver. And please stay with latest patch release, which is 0.4.6 at this point.

As to your questions, please see my answers below.

> 1. How to set keep_alive_timeout using clickhouse-jdbc-0.4.2-shaded ?

`keep_alive_timeout` is a server setting. The Java client simply respects the `Keep-Alive` header in response. Is there any specific reason you want to set this on client side?

> 2. What is the alternative of asProperties() in clickhouse-jdbc-0.4.2-shaded ?

You'd better use standard JDBC API ([example](https://github.com/ClickHouse/clickhouse-java/blob/dd91e1776b94a30e32fecf9c82354685295169a6/clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/ClickHouseDriver.java#L161)).

> 3. How to find the replacement of deprecated members alternative ?

If you need to pass server setting along with query, you may use `custom_settings` option as in the cheat sheet.

**Comment by shashank-gb** [comment]
Hi @zhicwu ,
Thanks for you quick response.

Question 2 is cleared.

For Q1, Do you mean setting a Keep-Alive from client side using `setKeepAliveTimeout` will nor change the server sides config ?
For Q3, What I meant is, Is there anywhere mention the alternative of deprecated methods. 
For e.g., A new Optimized method `xyz()` introduced in replace with method `abc()` which will deprecate in upcoming build. Is anywhere in documentation mentioned **use `xyz()` instead of `abc()`**

**Comment by zhicwu** [comment]
> For Q1, Do you mean setting a Keep-Alive from client side using setKeepAliveTimeout will nor change the server sides config ?

Yes, keep-alive can be only changed from server side configuration.

> For Q3, What I meant is, Is there anywhere mention the alternative of deprecated methods.

Unfortunately, we don't have the full list. After all, it's a complete rewrite followed by a few rounds of refactoring. Besides the cheat sheet, you may want to stay with standard JDBC API and check out examples at [here](/ClickHouse/clickhouse-java/tree/main/examples) for typical usage.

**Comment by shashank-gb** [comment]
Ok ,thanks 
I got answers for all the questions
Closing the issue.
